### PR TITLE
Added caching for Request results from database queries using javax.cache (JSR 107)

### PR DESCRIPTION
**Pull request by konradlang**
Added caching for Request results from database queries using javax.cache (JSR 107):

Added Spring Framework to support caching;
Added ehcache as cache provider;

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -7,47 +7,46 @@
...
+            <database.username>saher</database.username>
+            <database.password>saher</database.password>-->
 
         <!--<database.server>localhost</database.server>
         <database.port>5434</database.port>
         <database.name>negotiator</database.name>
         <database.username>negotiator</database.username>
         <database.password>negotiator</database.password>-->
 
-        <database.server>localhost</database.server>
```
Comment (RadovanTomik) [inline pom.xml:50]: please return it to localhost

**Review by RadovanTomik — CHANGES_REQUESTED** [review]
👍

File: pom.xml (hunk 1 above)
Comment (konradlang) [reply pom.xml:50]: done.

**Review by RadovanTomik — APPROVED** [review]
👍

**Comment by konradlang** [comment]
Marginal performance improvement, not merged.

**Pull request closed without merging**